contracts/governance/TimelockController.sol

```diff
@@ -346,22 +392,30 @@ contract TimelockController is AccessControl, IERC721Receiver, IERC1155Receiver
      */
     function _execute(address target, uint256 value, bytes calldata data) internal virtual {
         (bool success, ) = target.call{value: value}(data);
-        require(success, "TimelockController: underlying transaction reverted");
+        if (!success) {
```

Comment:
Nice call, implemented!

Reply:
Reverted since `functionCallWithValue` requires the balance of the caller to be above 1 wei:

```solidity
if (address(this).balance < value) {
    revert AddressInsufficientBalance(address(this));
}
```

Also, it uses `verifyCallResultFromTarget` under the hood which verifies the target is a contract.

Reply:
Why does `functionCallWithValue` require the call to have at least 1 wei ???

This looks like an error. In particular because `functionCall` calls `functionCallWithValue` under the hood.
Both the current implementation 
```
require(address(this).balance >= value, "Address: insufficient balance for call");
```
and the "custom error" implementation don't have this requirement

Reply:
> Why does functionCallWithValue require the call to have at least 1 wei ???

I made a mistake, I saw the error and jumped to conclusions immediately. Indeed the tests fail in TimelockController with an `AddressInsufficientBalance` but those are tests supposed to fail due to insufficient balance.

Still, the `functionCallWithValue` uses `verifyCallResutlFromTarget`, so it doesn't allow calls to non-contract addresses

Reply:
Yes. I always get confused by that part. I see the function doesn't check if the target is a contract before doing the call and I forget it does the check after :/